### PR TITLE
[kube-prometheus-stack] Allow to define metrics label on all kube-prometheus-stack monitors

### PR DESCRIPTION
**Pull request by jkroepke**
#### What this PR does / why we need it

With #1132 we can use multi-cluster dashboards with depends on the cluster label. While its possible to set the cluster label on federated metrics, its not possible to define this metric on the own metrics.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
please add it via more generic approach, where labels can be added freely.

otherwise if some other label would be needed a new option has to be added.

**Comment by jkroepke** [comment]
It should be more generic now.

**Review by monotek — CHANGES_REQUESTED** [review]
imho labels and targetLabels are differents things and should not be configureed via a single variable.

File: charts/kube-prometheus-stack/templates/prometheus/servicemonitors.yaml [hunk 1]
```diff
@@ -34,5 +37,9 @@ items:
       podTargetLabels:
 {{ toYaml .podTargetLabels | indent 8 }}
     {{- end }}
+    {{- if $.Values.clusterLabel }}
+    targetLabels:
+      - cluster
```
[the remark this answers is not in the document]
Comment (jkroepke) [reply charts/kube-prometheus-stack/templates/prometheus/servicemonitors.yaml]: thanks

**Comment by jkroepke** [comment]
About the `targetLabels` thing, before the PR, `targetLabels` was not configurable at all.

**Comment by oshamin** [comment]
I am looking for a solution to add "cluster" label to all ServiceMonitors. This PR could solve my problem, but it was not merged :(.

Are there any other simple options to do that?

**Pull request closed without merging**